File: DesafioConfitec-Web/src/app/operacoes.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class OperacoesService {
  uri = 'http://localhost:5000/api/operacoes';
  
  constructor(private http: HttpClient) { }

  somar(valor1, valor2) {
    return this.http.get(`${this.uri}/somar/${valor1}/${valor2}`);
  }

  subtrair(valor1: any, valor2: any) {
    return this.http.get(`${this.uri}/subtrair/${valor1}/${valor2}`);
  }

  multiplicar(valor1: any, valor2: any) {
    return this.http.get(`${this.uri}/multiplicar/${valor1}/${valor2}`);
  }

  dividir(valor1: any, valor2: any) {
    return this.http.get(`${this.uri}/dividir/${valor1}/${valor2}`);
  }
}

File: DesafioConfitec.Business/Service/OperacoesService.cs
using System;

namespace DesafioConfitec.Business.Service
{
    public class OperacoesService : IOperacoesService
    {
        public double Somar(double valor1, double valor2)
        {
            return valor1 + valor2;
        }

        public double Subtrair(double valor1, double valor2)
        {
            return valor1 - valor2;
        }
        public double Multiplicar(double valor1, double valor2)
        {
            return valor1 * valor2;
        }

        public double Dividir(double valor1, double valor2)
        {
            if (valor2 == 0)
                throw new DivideByZeroException();

            return valor1 / valor2;
        }
    }
}